spec/seeders/basic_data/life_cycle_seeder_spec.rb

```diff
@@ -0,0 +1,191 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+
+RSpec.describe BasicData::LifeCycleSeeder do
+  subject(:seeder) { described_class.new(seed_data) }
+
+  let(:initial_seed_data) { Source::SeedData.new({}) }
+  let(:second_seed_initial_data) { initial_seed_data }
+  let(:seed_data) { initial_seed_data.merge(Source::SeedData.new(data_hash)) }
+
+  context "with some life cycles defined" do
+    let(:data_hash) do
+      YAML.load <<~SEEDING_DATA_YAML
+        life_cycles:
+        - reference: :default_life_cycle_initiating
+          name: Initiating
+          type: Stage
+          color_name: :default_color_orange_dark
+        - reference: :default_life_cycle_ready_for_executing
+          name: Ready for Executing
+          type: Gate
+          color_name: :default_color_purple
+        - reference: :default_life_cycle_planning
+          name: Planning
+          type: Stage
+          color_name: :default_color_red_dark
+        - reference: :default_life_cycle_executing
+          name: Executing
+          type: Stage
+          color_name: :default_color_magenta_light
+        - reference: :default_life_cycle_closing
+          name: Closing
+          type: Stage
+          color_name: :default_color_green_yellow
+      SEEDING_DATA_YAML
+    end
+
+    shared_examples_for "creates the life_cycles seeds" do
+      it "creates the corresponding life cycles with the given attributes" do
+        expect(LifeCycle.count).to eq(5)
+        expect(Stage.find_by(name: "Initiating")).to have_attributes(
+          color_id: Color.find_by(name: "Orange (dark)").id
```

Comment:
I think it reads better without inversion:
```suggestion
          color: have_attributes(name: "Orange (dark)")
```